fip-3.md

```diff
@@ -0,0 +1,158 @@
+
+
+## Specification
+### Cancel Funds Request
+#### New end point: *cancel_funds_request*
+#### New contract action canclrequest
+##### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_request_id|Yes|Positive Int|Existing Funds request id|
+|max_fee|Yes|Max fee SUFs|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by /get_fee for correct value.|
+|tpid|Yes|FIO Address of TPID, See FIO Address validation rules|FIO Address of the wallet which generates this transaction. This FIO Address will be paid 10% of the fee.See FIO Protocol#TPIDs for details. Set to empty if not known.|
+|actor|Yes|the FIO account of the canceller|FIO pub account owning this payee FIO Address.|
+###### Example
+```
+{
+	"fio_request_id": "27",
+	"max_fee": 40000000000,
+	"tpid": ""
+    "actor":"edrtfgtrthg "
+}
+```
+##### Processing
+* Verify the request exists. Error if id is not present. 
+* Check that no payment is yet received for this request. If status is sent to blockchain then error.
+* Check that payee_fio_address is owned by actor.
+* verify that this payee key from the context table hashes to this actor.
+* require auth of the actor
+* verify that the fee for this does not exceed the max fee specified.
+* charge appropriate fee (this will be a bundled fee transaction, fee will be same as reject)
+* if there is not a status, create status to become cancelled.
+* if there is a status, and status is cancelled, or rejected, or sent to blockchain then error
+* if there is not a status then create a new cancelled status.
+* check that transaction does not exceed the max allowable tx size.
+* bump the ram allowance by 512
+* Return status json containing status (cancelled), and fee charged.
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid Request ID|Request ID not found|400|"fio_request_id"|Value sent in not found|"Invalid FIO Request ID"|
+|Invalid status|Request status is not pending|400|"fio_request_id"|"Only pending requests can be cancelled."|
+|Request ID|Request cannot be cancelled because status is sent to blockchain (2)|400|"fio_request_id"|request with status sent to blockchain cannot be cancelled |
+|Fee exceeds maximum|fee exceeded the specified max|400|"max_fee"|fee exceeds max_fee specified|"Fee exceeds maximum"|
+|Invalid TPID|TPID is not empty or contains invalid FIO address|400|"tpid"|Value sent in is not empty and not a valid FIO Address format|"TPID must be empty or valid FIO address"|
+|Invalid actor|payee pub key hashed to account does not match actor FIO Address owner of request does not match actor|403|||"Invalid Actor"|
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+||status|String|cancelled|
+||fee_collected|String|fee amount collected SUFs|
+###### Example
+```
+{
+	
+			"status": "cancelled",
+			"fee_collected": "0"		
+}
+```
```

Comment:
Let's clean up the whitespace on all json examples. I like to use tools like https://jsonlint.com/ for consistency.